src/App.tsx

```diff
@@ -1,26 +1,105 @@
-import React from 'react';
-import logo from './logo.svg';
+import React, {useState} from 'react';
+
 import './App.css';
 
-function App() {
-  return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.tsx</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
+
+interface Product {
+    category: string;
+    price: string;
+    stocked: boolean;
+    name: string;
+}
+
+interface AllProducts {
+    products: Array<Product>;
+};
```

Comment:
Please rename interface to `AppProps`, because `products` are props that are passed to <App/> component - that's why "AppProps":
```diff
- interface AllProducts {
+ interface AppProps {
    products: Array<Product>;
};
```